libs/core/src/main/java/org/opensearch/core/common/transport/TransportAddress.java

```diff
@@ -162,4 +164,22 @@ public String toString() {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.value(toString());
     }
+
+    /**
+    * Converts a string in the format [hostname/ip]:[port] into a transport address.
+    * @throws UnknownHostException if the hostname or ip address is invalid
+    */
+    public static TransportAddress fromString(String address) throws UnknownHostException {
```

Comment:
You don't control the toString() you are using ```TransportAddress.toString()``` which calls ```NetworkAddress.format(address);``` . If toString() implementation changes, your fromString() will break.